six/modules/c++/cphd/source/CPHDXMLControl.cpp

```diff
@@ -99,7 +107,7 @@ std::unique_ptr<xml::lite::Document> CPHDXMLControl::toXML(
 
 std::unique_ptr<xml::lite::Document> CPHDXMLControl::toXMLImpl(const Metadata& metadata)
 {
-    return getParser(metadata.getUri())->toXML(metadata);
+    return getParser(VERSION_URI_MAP.find(metadata.getVersion())->second)->toXML(metadata);
```

Comment:
If user provided an invalid version, then accessing the result of `find` will cause UB.